bandersnatch/multiexp.go

```diff
@@ -367,12 +369,40 @@ func msmReduceChunkPointAffine(p *PointProj, c int, chChunks []chan PointProj) *
 	return p
 }
 
+func msmReduceChunkPointAffineDMA(p *PointProj, c int, chChunks []PointProj) *PointProj {
+	var _p PointProj
+	totalj := chChunks[len(chChunks)-1]
+	_p.Set(&totalj)
+	for j := len(chChunks) - 2; j >= 0; j-- {
+		for l := 0; l < c; l++ {
+			_p.Double(&_p)
+		}
+		totalj := chChunks[j]
+		_p.Add(&_p, &totalj)
+	}
+
+	p.Set(&_p)
+
+	return p
+}
+
 func msmProcessChunkPointAffine(chunk uint64,
 	chRes chan<- PointProj,
 	buckets []PointProj,
 	c uint64,
 	points []PointAffine,
 	scalars []fr.Element) {
+	var total PointProj
+	msmProcessChunkPointAffineDMA(chunk, &total, buckets, c, points, scalars)
+	chRes <- total
+}
+
+func msmProcessChunkPointAffineDMA(chunk uint64,
```

Comment:
The TL;DR in the multiexp is that a channel-based solution to the "map->reduce"  created many allocations regarding creating so many channels.

I noticed that the way the work was structured, it could be directly mapped to a slice index avoiding all that extra work.

The "reduce" code is backward compatible since there's another code that expects to receive the result in a channel.  (Code above this comment).